test/performance/search/views/view_kmer_hash_benchmark.cpp

```diff
@@ -104,8 +103,22 @@ static void naive_kmer_hash(benchmark::State & state)
 
     for (auto _ : state)
     {
-        for (auto h : seq | seqan3::views::naive_kmer_hash(k))
+        // creates a initial subrange covering exactly 'k-1' characters
+        auto subRangeBegin = begin(seq);
+        auto subRangeEnd   = subRangeBegin;
+        for (size_t i{1}; i < k; ++i && subRangeEnd != end(seq))
+            ++subRangeEnd;
```

Comment:
```suggestion
        auto subRangeEnd   = std::ranges::next(subRangeBegin, k - 1, end(seq));
```

Should be equivalent. Copies `subRangeBegin`, increments it `k-1` times, and uses `end` as boundary.
The only thing I'm not sure about is the `k-1` :D

Reply:
Awesome, much nicer solution!